Jenkinsfile

```diff
@@ -0,0 +1,128 @@
+#!groovy
+
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+
+PLUGIN_NAME = "sample-angular-app"
+PLUGIN_ID = 'org.zowe.zlux.sample-angular-app'
+
+GITHUB_SSH_KEY = "zlux-jenkins" // This is required for git+ssh npm dependencies
+
+
+node("zlux-agent") {
+
+  def lib = library("jenkins-library").org.zowe.jenkins_shared_library
+  def pipeline = lib.pipelines.generic.GenericPipeline.new(this)
+  pipeline.admins.add("dnikolaev", "sgrady")
+
+  def baseBranch = env.CHANGE_TARGET? env.CHANGE_TARGET: env.BRANCH_NAME  
+  // A workaround to disable default checkout
+  (plugin_scm, scm) = [scm, null]
+
+  def zoweManifestURL = \
+      "https://raw.githubusercontent.com/zowe/zowe-install-packaging" +
+      "/${baseBranch}/manifest.json.template"
+  def zoweManifestText = httpRequest(url: zoweManifestURL).content.replace("{BUILD_NUMBER}", "-")
+
+  pipeline.setup(
+    packageName: PLUGIN_ID,
+    version: readJSON(text: zoweManifestText)["version"],
+    github: [
+      email : lib.Constants.DEFAULT_GITHUB_ROBOT_EMAIL,
+      usernamePasswordCredential: lib.Constants.DEFAULT_GITHUB_ROBOT_CREDENTIAL
+    ],
+    artifactory: [
+      url: lib.Constants.DEFAULT_ARTIFACTORY_URL,
+      usernamePasswordCredential : lib.Constants.DEFAULT_ARTIFACTORY_ROBOT_CREDENTIAL
+    ],
+    pax: [
+      sshHost: lib.Constants.DEFAULT_PAX_PACKAGING_SSH_HOST,
+      sshPort: lib.Constants.DEFAULT_PAX_PACKAGING_SSH_PORT,
+      sshCredential: lib.Constants.DEFAULT_PAX_PACKAGING_SSH_CREDENTIAL,
+      remoteWorkspace: lib.Constants.DEFAULT_PAX_PACKAGING_REMOTE_WORKSPACE,
+    ],
+  )
```

Comment:
With https://github.com/zowe/jenkins-library/issues/17, a new option `skipCheckout: true` is available for `setup()` method.